### PR TITLE
Add initial cpp module

### PR DESCRIPTION
**Pull request by gitbuda**
- [x] Add build infrastructure (cpp only).

### CODE REVIEW

File: cpp/connectivity_module.cpp [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+      continue;
+
+    // run bfs from current vertex
+    std::queue<int64_t> q;
+    q.push(vertex_id.as_int);
+    vertex_component[vertex_id.as_int] = curr_component;
+    while (!q.empty()) {
+      mgp_vertex *v = mgp_graph_get_vertex_by_id(graph, {q.front()}, memory);
+      if (v == nullptr) {
+        mgp_vertices_iterator_destroy(vertices_iterator);
```
[the remark this answers is not in the document]
Comment (gitbuda) [reply cpp/connectivity_module.cpp]: All add this today!

File: cpp/connectivity_module/connectivity_module.cpp [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+static void weak(const mgp_list *args, const mgp_graph *graph,
+                 mgp_result *result, mgp_memory *memory) {
+  std::unordered_map<int64_t, int64_t> vertex_component;
+  mgp_vertices_iterator *vertices_iterator =
+      mgp_graph_iter_vertices(graph, memory);
+  if (vertices_iterator == nullptr) {
+    mgp_result_set_error_msg(result, "Not enough memory");
+    return;
+  }
+  utils::OnScopeExit([vertices_iterator] {
```
Comment (antonio2368) [inline cpp/connectivity_module/connectivity_module.cpp:18]: &vertices_iterator so the check makes sense

File: cpp/connectivity_module/connectivity_module.cpp [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+    std::queue<int64_t> q;
+    q.push(vertex_id.as_int);
+    vertex_component[vertex_id.as_int] = curr_component;
+    while (!q.empty()) {
+      mgp_vertex *v = mgp_graph_get_vertex_by_id(graph, {q.front()}, memory);
+      if (v == nullptr) {
+        mgp_result_set_error_msg(result, "Not enough memory");
+        return;
+      }
+      utils::OnScopeExit([v] {
```
Comment (antonio2368) [inline cpp/connectivity_module/connectivity_module.cpp:42]: &v

File: cpp/CMakeLists.txt [hunk 4]
```diff
@@ -0,0 +1,32 @@
+# Memgraph Query Modules CMake C++ configuration.
+
+cmake_minimum_required(VERSION 3.8)
+
+project(memgraph-query-modules)
```
Comment (antonio2368) [inline cpp/CMakeLists.txt:5]: memgraph-mage

**Review by antonio2368 — APPROVED** [review]

**Pull request merged**